### PR TITLE
spglib 2.1.0 (new formula)

### PR DESCRIPTION
**Pull request by LecrisUT**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I haven't had a homebrew environment in a while so I would appreciate some help getting up to speed on setting one up in a container to do the local testing as needed. Also [upstream issue](https://github.com/spglib/spglib/issues/352)

### CODE REVIEW

**Comment by LecrisUT** [comment]
What is `Commit style` check checking for? The commits should just add some simple text to the body?

File: Formula/s/spglib.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+             "-S", ".",
+             "-B", build_dir,
+             *common_args,
+             "-DSPGLIB_SHARED_LIBS=#{shared_lib_type ? "ON" : "OFF"}",
+             *std_cmake_args
+      system "cmake", "--build", build_dir
+      system "ctest",
+             "-j", ENV.make_jobs,
+             "--test-dir", build_dir,
+             "--output-on-failure"
```
Comment (SMillerDev) [inline Formula/s/spglib.rb:27]: We don't run tests in packages.

Comment (LecrisUT) [reply Formula/s/spglib.rb:27]: Ok, I've seen a few of them that run. Any advice on how to run these upstream or equivalent, maybe by maintaining an upstream tap? It does pick up on some errors that we should address upstream.

Comment (SMillerDev) [reply Formula/s/spglib.rb:27]: Run it in upstream CI?

Comment (LecrisUT) [reply Formula/s/spglib.rb:27]: Yes, if possible to setup an upstream tap to run the same GitHub actions as here, but with upstream testing enabled, that would be ideal. I haven't looked into how to set it up, but from another discussion I understood that would be possible.

File: Formula/s/spglib.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      build_dir = "build_#{shared_lib_type ? "shared" : "static"}"
+      system "cmake",
+             "-S", ".",
+             "-B", build_dir,
+             *common_args,
+             "-DSPGLIB_SHARED_LIBS=#{shared_lib_type ? "ON" : "OFF"}",
+             *std_cmake_args
+      system "cmake", "--build", build_dir
+      system "cmake", "--install", build_dir
+    end
```
Comment (SMillerDev) [inline Formula/s/spglib.rb:25]: Can you just add this twice? It's much easier to read than the loop.

Comment (LecrisUT) [reply Formula/s/spglib.rb:25]: Yes, but wouldn't it be a bit more cumbersome to maintain? Also in the case of the test?

Comment (SMillerDev) [reply Formula/s/spglib.rb:27]: Not really, since it would be much clearer what is happening.

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**